src/test/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistencyTest.java

```diff
@@ -0,0 +1,109 @@
+package com.atlassian.db.replica.api;
+
+import com.atlassian.db.replica.internal.aurora.AuroraReplicaNode;
+import com.atlassian.db.replica.spi.ReplicaConnectionPerUrlProvider;
+import com.atlassian.db.replica.spi.ReplicaConsistency;
+import com.atlassian.db.replica.spi.SuppliedCache;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class AuroraMultiReplicaConsistencyTest {
+    private Collection<Connection> replicas;
+
+    @Mock
+    private Connection supplierConnection;
+    @Mock
+    private SuppliedCache<Collection<Database>> discoveredReplicasCache;
+
+    private AuroraMultiReplicaConsistency sut;
+
+    private void mockReplicaConnections(int count) {
+        replicas = new LinkedList<>();
```

Comment:
Why reusing replicas between tests? The method could return a new copy the risk of flakiness will decrease.

Reply:
Does noone use it?

Reply:
I see it's shared between the test cases.

Reply:
will make the `replicas` a local variable for each test